src/base/Dispatcher.sol

```diff
@@ -209,11 +211,14 @@ abstract contract Dispatcher is
                             permitSingle := inputs.offset
                         }
                         bytes calldata data = inputs.toBytes(6); // PermitSingle takes first 6 slots (0..5)
-                        try PERMIT2.permit(msgSender(), permitSingle, data) {}
-                        catch (bytes memory reason) {
-                            output = reason;
-                            success = false;
-                        }
+                        (success, output) = address(PERMIT2).call(
+                            abi.encodeWithSignature(
```

Comment:
may be a tiny one, does abi.encodeCall better ?

Reply:
Yes encodeCall provides type safety.

Reply:
we can't use that as there's multiple `permit` function

Reply:
eg. this is the error `Member "permit" not unique after argument-dependent lookup in type(contract IAllowanceTransfer)`

Reply:
will merge for now, though if there's any follow-up action required, will do follow-up PR to resolve the comment